source/features/wait-for-attachments.tsx

```diff
@@ -31,11 +31,18 @@ function toggleSubmitButtons({target, type}: Event): void {
 	}
 }
 
-function init(): void {
-	document.addEventListener('upload:setup', toggleSubmitButtons, true);
+function init(): Deinit {
+	document.addEventListener('upload:setup', toggleSubmitButtons, {capture: true});
 	document.addEventListener('upload:complete', toggleSubmitButtons);
 	document.addEventListener('upload:error', toggleSubmitButtons);
 	document.addEventListener('upload:invalid', toggleSubmitButtons);
+
+	return () => {
+		document.removeEventListener('upload:setup', toggleSubmitButtons, {capture: true});
+		document.removeEventListener('upload:complete', toggleSubmitButtons);
+		document.removeEventListener('upload:error', toggleSubmitButtons);
+		document.removeEventListener('upload:invalid', toggleSubmitButtons);
```

Comment:
In a later cleanup PR, I think we should just create the `signal` helper I mentioned before. This is so error-prone and I don't wanna see it.

The code would instead look like:

```js
import on from '../heh'

function init(signal: AbortSignal): void {
 	on(document, 'upload:setup', toggleSubmitButtons, {signal, capture: true});
	on(document, 'upload:complete', toggleSubmitButtons, {signal});
 	on(document, 'upload:error', toggleSubmitButtons, {signal});
 	on(document, 'upload:invalid', toggleSubmitButtons, {signal});
}
```

and it would actually kinda mirror `delegate` (which would also benefit from https://github.com/fregante/delegate-it/issues/26)

Reply:
Why not make that helper return a wrapper function containing a `removeEventListener` instead? It'd be more consistent with the way `delegate` works now imo.